### PR TITLE
Fix a couple of flaky suites

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler 

This closes #6357 and #6540

File: src/Compilers/Server/PortableServer/CoreClrCompilerServerHost.cs [hunk 1]
```diff
@@ -15,17 +15,15 @@ namespace Microsoft.CodeAnalysis.CompilerServer
...
-
-        public override Func<string, MetadataReferenceProperties, PortableExecutableReference> AssemblyReferenceProvider => _assemblyReferenceProvider;
+        public override Func<string, MetadataReferenceProperties, PortableExecutableReference> AssemblyReferenceProvider { get; }
 
         internal CoreClrCompilerServerHost(string clientDirectory)
             :base(clientDirectory : clientDirectory, sdkDirectory: null)
         {
-
+            AssemblyReferenceProvider = (path, properties) => new CachingMetadataReference(path, properties);
+            AnalyzerAssemblyLoader = CoreClrAnalyzerAssemblyLoader.CreateAndSetDefault();
```
Comment (AlekseyTs) [inline src/Compilers/Server/PortableServer/CoreClrCompilerServerHost.cs]: Are changes in this file necessary to address the problem. If not, consider moving them to a separate PR.

Comment (jaredpar) [inline src/Compilers/Server/PortableServer/CoreClrCompilerServerHost.cs]: Okay.  This was done in response to feedback I got on a separate PR.  I forgot to merge that separately.

File: src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs [hunk 2]
```diff
@@ -229,6 +202,44 @@ internal NamedPipeClientConnection(ICompilerServerHost compilerServerHost, strin
...
+            if (!ClientAndOurIdentitiesMatch(_pipeStream))
+            {
+                throw new Exception("Client identity does not match server identity.");
+            }
+        }
+
+        /// <summary>
+        /// Does the client of "pipeStream" have the same identity and elevation as we do?
+        /// </summary>
+        private static bool ClientAndOurIdentitiesMatch(NamedPipeServerStream pipeStream)
```
Comment (AlekseyTs) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: Were this and the function below simply moved or were they also changed in some way?

Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: The content was just moved.

File: src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs [hunk 3]
```diff
@@ -17,18 +17,14 @@
 
 namespace Microsoft.CodeAnalysis.CompilerServer
 {
-    internal static class VBCSCmopiler
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs]: Jared can't spell :frowning:

File: src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs [hunk 4]
```diff
@@ -229,6 +202,44 @@ internal NamedPipeClientConnection(ICompilerServerHost compilerServerHost, strin
             }
         }
 
+        protected override void ValidateBuildRequest(BuildRequest request)
```
Comment (AlekseyTs) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: It doesn't look like this function is called anywhere.

Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: Doh.  Will fix.

Comment (AlekseyTs) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: Please add a unit-test if possible.

Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: Added a test for this.

File: src/Compilers/Server/PortableServer/CoreClrCompilerServerHost.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Server/PortableServer/CoreClrCompilerServerHost.cs]: Unfortunately i've been unable to remove this.  I tried rebasing and manually editing this change out but I can't get it correct.  I'll be more dilligent about this in the future :frowning:

**Comment by jaredpar** [comment]
Responded to PR feedback.

Ping @dotnet/roslyn-compiler

File: src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs (hunk 4 above)
Comment (AlekseyTs) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: I was actually hoping to have a unit-test for the logic in this particular override rather than a unit-test that ensures that the method is called.

**Comment by AlekseyTs** [comment]
LGTM, modulo a missing test for ValidateBuildRequest override.

File: src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs (hunk 4 above)
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: I'm not sure I can test this given our current Jenkins setup.  In order to fail this test I need the client and server pipe to have different identities.  Essentially I need to have two users accounts available.  

Let me chat with Andy when I get back in town.  I'll see if we can devise a way to get some coverage here.  Even if it's only in specific runs where multiple accounts are made available to the xunit process.

**Comment by jaredpar** [comment]
Filed #6741 for the test issue as discussed.
